text/0763-asset-importing-spec.md

```diff
@@ -0,0 +1,200 @@
+---
+Stage: Accepted
+Start Date: 2021-08-13
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js, Ember CLI
+RFC PR: https://github.com/emberjs/rfcs/pull/763
+---
+
+<!---
+Directions for above:
+
+Stage: Leave as is
+Start Date: Fill in with today's date, YYYY-MM-DD
+Release Date: Leave as is
+Release Versions: Leave as is
+Relevant Team(s): Fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies
+RFC PR: Fill this in with the URL for the Proposal RFC PR
+-->
+
+# Asset Import Spec
```

Comment:
When it comes to static assets, such as images, it is  common for them to be "internationalized". Do you have thoughts on how this proposal could play with that?

For example, let's imagine i have the following localized versions of `hello.png`:

* `hello.en-us.png`
* `hello.fr-fr.png`

Where the form is something like: `hello.<%locale%>.png` or maybe `<%locale%>/hello.png`

Reply:
In JS this would take advantage of dynamic import with a template string literal argument (which is already allowed in the embroider spec) or the `importSync` macro (which works in classic apps too).

In CSS, I don't think we would support it directly and you'd need to either work in JS or use a preprocessor on the CSS that expands outs the substitutions. I'm pretty sure CSS vars can't be interpolated into URLs, but if they can we could respect that.

Reply:
So TL;DR `importSync` would be used to add the right dependency from JS -> CSS?